hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -188,10 +188,14 @@ SqlResult execute(CreateIndexPlan plan) {
 
     SqlResult execute(CreateJobPlan plan, List<Object> arguments) {
         List<Object> args = prepareArguments(plan.getParameterMetadata(), arguments);
+        boolean isStreamingJob = plan.isInfiniteRows();
         JobConfig jobConfig = plan.getJobConfig()
                 .setArgument(SQL_ARGUMENTS_KEY_NAME, args)
                 .setArgument(KEY_SQL_QUERY_TEXT, plan.getQuery())
-                .setArgument(KEY_SQL_UNBOUNDED, plan.isInfiniteRows());
+                .setArgument(KEY_SQL_UNBOUNDED, isStreamingJob);
```

Comment:
What about the other plans? Select/DML? Those also can be streaming.

Reply:
Is there any description of this change apart from issue title?

Do we want one-shot streaming SQL queries (that is not executed using `CREATE JOB`) to be suspended? From the user point of view such query probably failed. I am not sure if they would be able to "reconnect" to it after restart to get more select results. For DML that might make more sense, but user would not know/expect that failed INSERT/UPDATE is still sleeping somewhere to be resumed. Some would also have to clean snapshots.

All-in-all, for me it makes sense to enable `suspend-on-failure` only for cases when user explicitly wanted something more durable, i.e. used `CREATE JOB`.

There is also a technical reason: normal queries are light jobs which cannot be suspended.

Reply:
@krzysztofslusarski it's applied for CREATE JOB statement, and it should work fine for any plans covered by `CreateJobPlan`

Reply:
@k-jamroz [link](https://hazelcast.atlassian.net/wiki/spaces/PM/pages/4318986241/Jet+Job+Management+Improvements)